src/contracts/atlas/GasAccounting.sol

```diff
@@ -16,149 +18,175 @@ abstract contract GasAccounting is SafetyLocks {
         address _simulator
     )
-        SafetyLocks(_escrowDuration, _factory, _verification, _gasAccLib, _safetyLocksLib, _simulator)
+        SafetyLocks(_escrowDuration, _factory, _verification, _simulator)
     { }
 
     // ---------------------------------------
     //          EXTERNAL FUNCTIONS
     // ---------------------------------------
 
     // Returns true if Solver status is Finalized and the caller (Execution Environment) is in surplus
-    // NOTE: This was a view function until logic got moved to delegatecalled contract - still should only be view
-    function validateBalances() external returns (bool valid) {
-        valid = ledgers[uint256(Party.Solver)].status == LedgerStatus.Finalized && _isInSurplus(msg.sender);
-    }
-
-    function deposit(Party party) external payable {
-        (bool success,) = GAS_ACC_LIB.delegatecall(abi.encodeWithSelector(GasAccountingLib.deposit.selector, party));
-        if (!success) revert GasAccountingLibError();
-    }
-
-    function contribute(Party recipient) external payable {
-        (bool success,) =
-            GAS_ACC_LIB.delegatecall(abi.encodeWithSelector(GasAccountingLib.contribute.selector, recipient));
-        if (!success) revert GasAccountingLibError();
+    function validateBalances() external view returns (bool valid) {
+        return solver == SOLVER_FULFILLED;
     }
 
-    function contributeTo(Party donor, Party recipient, uint256 amt) external {
-        (bool success,) = GAS_ACC_LIB.delegatecall(
-            abi.encodeWithSelector(GasAccountingLib.contributeTo.selector, msg.sender, donor, recipient, amt)
-        );
-        if (!success) revert GasAccountingLibError();
+    function contribute() external payable {
+        if (lock != msg.sender) revert InvalidExecutionEnvironment(lock);
+        _contribute();
     }
 
-    function requestFrom(Party donor, Party recipient, uint256 amt) external {
-        (bool success,) = GAS_ACC_LIB.delegatecall(
-            abi.encodeWithSelector(GasAccountingLib.requestFrom.selector, msg.sender, donor, recipient, amt)
-        );
-        if (!success) revert GasAccountingLibError();
+    function borrow(uint256 amount) external payable {
+        if (lock != msg.sender) revert InvalidExecutionEnvironment(lock);
+        if (_borrow(amount)) {
+            SafeTransferLib.safeTransferETH(msg.sender, amount);
+        } else {
+            revert InsufficientAtlETHBalance(address(this).balance, amount);
+        }
     }
 
-    function finalize(Party party, address partyAddress) external {
-        (bool success,) =
-            GAS_ACC_LIB.delegatecall(abi.encodeWithSelector(GasAccountingLib.finalize.selector, party, partyAddress));
-        if (!success) revert GasAccountingLibError();
+    function shortfall() external view returns (uint256) {
+        uint256 deficit = claims + withdrawals;
+        uint256 _deposits = deposits;
+        return (deficit > _deposits ? deficit - _deposits : 0) + 1;
```

Comment:
Can't find any code where that relies on this `+ 1` (e.g. using > instead of >=) Any ideas or can we just remove the `+1`?

Reply:
You're right - let's do it.